tests/pretty_format_kotlin_test.py

```diff
@@ -57,9 +71,56 @@ def test__download_kotlin_formatter_jar(ensure_download_possible, version):  # n
         ("NotPrettyFormattedFixed.kt", 0),
     ),
 )
-def test_pretty_format_kotlin(undecorate_method, filename, expected_retval):
+def test_pretty_format_kotlin_ktlint(undecorate_method, filename, expected_retval):
     assert undecorate_method([filename]) == expected_retval
 
 
-def test_pretty_format_kotlin_autofix(tmpdir, undecorate_method):
-    run_autofix_test(tmpdir, undecorate_method, "NotPrettyFormatted.kt", "NotPrettyFormattedFixed.kt")
+@pytest.mark.parametrize(
+    ("filename", "expected_retval", "extra_parameters"),
+    (
+        ("Invalid.kt", 1, []),
+        ("PrettyFormatted.kt", 0, []),
+        ("NotPrettyFormatted.kt", 1, []),
+        ("NotPrettyFormattedFixedKtlint.kt", 0, []),
+        # ktfmt dropbox style (same as kotlinlang in this example)
+        ("NotPrettyFormattedFixedKtfmtDropbox.kt", 1, ["--ktfmt"]),
+        ("NotPrettyFormattedFixedKtfmtDropbox.kt", 0, ["--ktfmt", "--ktfmt-style=dropbox"]),
+        ("NotPrettyFormattedFixedKtfmtDropbox.kt", 1, ["--ktfmt", "--ktfmt-style=google"]),
+        ("NotPrettyFormattedFixedKtfmtDropbox.kt", 0, ["--ktfmt", "--ktfmt-style=kotlinlang"]),
+        # ktfmt google style
+        ("NotPrettyFormattedFixedKtfmtGoogle.kt", 0, ["--ktfmt"]),
+        ("NotPrettyFormattedFixedKtfmtGoogle.kt", 1, ["--ktfmt", "--ktfmt-style=dropbox"]),
+        ("NotPrettyFormattedFixedKtfmtGoogle.kt", 0, ["--ktfmt", "--ktfmt-style=google"]),
+        ("NotPrettyFormattedFixedKtfmtGoogle.kt", 1, ["--ktfmt", "--ktfmt-style=kotlinlang"]),
+        # ktfmt kotlinlang style (same as dropbox in this example)
+        ("NotPrettyFormattedFixedKtfmtKotlinlang.kt", 1, ["--ktfmt"]),
+        ("NotPrettyFormattedFixedKtfmtKotlinlang.kt", 0, ["--ktfmt", "--ktfmt-style=dropbox"]),
+        ("NotPrettyFormattedFixedKtfmtKotlinlang.kt", 1, ["--ktfmt", "--ktfmt-style=google"]),
+        ("NotPrettyFormattedFixedKtfmtKotlinlang.kt", 0, ["--ktfmt", "--ktfmt-style=kotlinlang"]),
+        # ktfmt other files
+        ("NotPrettyFormatted.kt", 1, ["--ktfmt"]),
+        ("Invalid.kt", 1, ["--ktfmt"]),
+    )
+)
+def test_pretty_format_kotlin(undecorate_method, filename, expected_retval, extra_parameters):
+    assert undecorate_method(extra_parameters + [filename]) == expected_retval
```

Comment:
It might be worth splitting this test in 
* `test_pretty_format_kotlin_ktlint` (previous test)
* `test_pretty_format_kotlin_ktfmt`
* `test_pretty_format_kotlin_ktfmt_with_dropbox_style`
* ...

Tests are going to be somewhat easier to handle and more importantly you won't need to pass `--ktfmt` argument all the times and for the handled styles then `--ktfmt-style=...` is not needed (as the test can add it)